storage/Action.h

```diff
@@ -24,10 +24,10 @@ namespace storage
 
 	    virtual ~Base() {}
 
-	    virtual Text text(const Actiongraph& actiongraph, bool doing) const = 0;
-	    virtual void commit(const Actiongraph& actiongraph) const {} // = 0; TODO
+	    virtual Text text(const Actiongraph::Impl& actiongraph, bool doing) const = 0;
```

Comment:
Action.h is not part of the public interface. Maybe it will be in the future (see two TODOs in the code).